pkg/service/s3_context.go

```diff
@@ -159,26 +159,33 @@ func removeLeadingSlash(s string) string {
 }
 
 // CleanDir cleans the directory with the given name.
-func (s *S3Context) CleanDir(name string) {
-	path := s.Path + "/" + name
+func (s *S3Context) CleanDir(name string) error {
+	path := removeLeadingSlash(s.Path + "/" + name)
```

Comment:
This line is actually fix

Reply:
Needs to be tested on both MinIO and S3.

Reply:
In the ecosystem project